src/models/user.js

```diff
@@ -1,15 +1,54 @@
 import crypto from 'crypto'
 import { Model } from 'sequelize'
+import jwtBuilder from '../JWTBuilder'
+import InvalidCredentialsError from '../errors/InvalidCredentialsError'
 
-export default (sequelize, DataTypes) => {
+export function buildUserModel (sequelizeModels) {
   class User extends Model {
-    async passwordMatches (value, models = sequelize.models) {
+    static async signin (username, password) {
+      const user = await User.findByUsername(username)
+
+      await user.validatePassword(password)
+
+      const jwt = await user.generateJWT()
+
+      return { user, jwt }
+    }
+
+    static async findByUsername (username) {
+      const user = await User.findOne({ where: { username } })
+
+      if (!user) {
+        throw new InvalidCredentialsError()
```

Comment:
Hmmm... What if I want to use this method outside the sign in process? The exception should be thrown in you `signin` method.

Reply:
I thought the user not being found was an error condition, and since we're using the username to find it, it'd be an invalid credential error. Nonetheless, I can return `null` and throw the exception in `signin`.

Reply:
Yes, that's how the other methods work (findByPk, findOne, findAll), right? So, I'd expect this method to work the same.